### PR TITLE
[OCPCLOUD-1201] Add periodics for TechPreviewNoUpgrade clusters

### PR DESCRIPTION
**Pull request by JoelSpeed**
As part of the Cloud Controller Manager project, we are making some significant changes that will at first, be under the `TechPreviewNoUpgrade` feature set.
To gain confidence in the new cloud controller managers, we would like to add some signal to the release informing jobs that the TPNU clusters are still viable and healthy.

This PR adds periodics for the parallel and serial jobs for AWS, Azure and OpenStack, the three platforms we are initially targeting for the CCM project.

/assign @deads2k

### CODE REVIEW

**Comment by JoelSpeed** [comment]
/hold

This isn't actually doing what I want it to do right now, it is putting the FG in after install where we need it in the manifests before install

**Comment by JoelSpeed** [comment]
/retest

Build01 was broken yesterday so the jobs weren't starting correctly, I'm told this is now fixed

**Comment by JoelSpeed** [comment]
Looks like this is now working as expected. The tests have failed due to a known issue with the MCO, grabbed the below from the `machine-config` `ClusterOperator`
```
"extension": {
                    "master": "pool is degraded because nodes fail with \"3 nodes are reporting degraded status on sync\": \"Node ip-10-0-172-141.us-east-2.compute.internal is reporting: \\\"machineconfig.machineconfiguration.openshift.io \\\\\\\"rendered-master-0453ee164ce7ad45cc973e0bae480a25\\\\\\\" not found\\\", Node ip-10-0-139-14.us-east-2.compute.internal is reporting: \\\"machineconfig.machineconfiguration.openshift.io \\\\\\\"rendered-master-0453ee164ce7ad45cc973e0bae480a25\\\\\\\" not found\\\", Node ip-10-0-210-207.us-east-2.compute.internal is reporting: \\\"machineconfig.machineconfiguration.openshift.io \\\\\\\"rendered-master-0453ee164ce7ad45cc973e0bae480a25\\\\\\\" not found\\\"\"",
                    "worker": "all 3 nodes are at latest configuration rendered-worker-2b21538e8756e810dba78aa2edace63d"
                },
```
I believe the MCO team have a fix in place for this that is currently under review

**Comment by JoelSpeed** [comment]
/hold cancel

This is now working as intended, but we know this is broken for now

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml
Comment (deads2k) [inline ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:229]: not speaking cron, how frequent is this?

**Comment by deads2k** [comment]
Not sure I have rights here actually, let's try.

/approve

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml
Comment (JoelSpeed) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:229]: It is every 8 hours, copied from the existing openstack jobs which this would synchronise with. Will talk to the release team about why openstack is so frequent and whether being in sync is going to cause issues

File: ci-operator/step-registry/openshift/e2e/openstack/techpreview/openshift-e2e-openstack-techpreview-workflow.yaml
Comment (ravisantoshgudimetla) [inline ci-operator/step-registry/openshift/e2e/openstack/techpreview/openshift-e2e-openstack-techpreview-workflow.yaml:14]: No serial test-suite for OpenStack?

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml
Comment (ravisantoshgudimetla) [inline ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:242]: Ohh, so we just have periodics for serial suite?

**Comment by stbenjam** [comment]
- Do you only want them running on cron or do you want feedback on each nightly release? If the latter you probably want to add them to the release controller as optional informers: https://docs.ci.openshift.org/docs/architecture/release-gating/

- The OWNERS files for the steps seem to be missing the PR author, should you be including yourself to maintain these new steps?

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml
Comment (stbenjam) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:229]: https://crontab.guru

File: ci-operator/step-registry/ipi/openstack/pre/techpreview/ipi-openstack-pre-techpreview-chain.yaml
Comment (stbenjam) [inline ci-operator/step-registry/ipi/openstack/pre/techpreview/ipi-openstack-pre-techpreview-chain.yaml:8]: This comment seems to be incorrect

File: ci-operator/step-registry/ipi/conf/openstack/techpreview/ipi-conf-openstack-techpreview-chain.yaml
Comment (stbenjam) [inline ci-operator/step-registry/ipi/conf/openstack/techpreview/ipi-conf-openstack-techpreview-chain.yaml:15]: This step documentation doesn't seem to be accurate as to the purpose of this chain

**Review by stbenjam — CHANGES_REQUESTED** [review]
The comments on each of the new steps, chains and workflows should be updated to reflect their actual purposes

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml
Comment (stbenjam) [inline ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:228]: Why is parallel in the test name? It's assumed that's what's running unless otherwise specified.

Comment (JoelSpeed) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:228]: I've copied this from the test above, `e2e-openstack-parallel` on line 221 in this diff. Not sure why it's called that but this is just keeping consistency with the rest of the openstack tests

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml
Comment (JoelSpeed) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:242]: We have serial and the regular tests (parallel?), the block above this should be for the openstack regular tests

File: ci-operator/step-registry/openshift/e2e/openstack/techpreview/openshift-e2e-openstack-techpreview-workflow.yaml
Comment (JoelSpeed) [reply ci-operator/step-registry/openshift/e2e/openstack/techpreview/openshift-e2e-openstack-techpreview-workflow.yaml:14]: For openstack, this appears to be overriden by an environment variable within the job configuration itself. I'm just trying to be consistent with existing openstack jobs by copying that pattern

**Review by JoelSpeed — COMMENTED** [review]
Will go through and fix up all the comments on the files today, that was just an omission while I was copy/pasting

> Do you only want them running on cron or do you want feedback on each nightly release? If the latter you probably want to add them to the release controller as optional informers: https://docs.ci.openshift.org/docs/architecture/release-gating/

I'll look into that link thank you, I think we want both the cron and release optional informers

> The OWNERS files for the steps seem to be missing the PR author, should you be including yourself to maintain these new steps?

I deliberately didn't do that initially since this isn't really my remit. I've been asked to add these by archs because we are making the tech preview more interesting this release (with the CCMs), so it was suggested that it would be good in general to have this, rather than it being something specifically tied to the CCM project which I own. My thought here was that this should be owned in general by the release teams who own the rest of the release informing jobs.

If that's not the correct assumption, I'll update the PR to include myself as an owner on all of these as well

**Comment by JoelSpeed** [comment]
@stbenjam @ravisantoshgudimetla I think I've addressed and replied to all the comments as appropriate, PTAL when you have a moment

**Comment by JoelSpeed** [comment]
The MCO failure here will be fixed once https://github.com/openshift/machine-config-operator/pull/2668 is merged

**Comment by JoelSpeed** [comment]
To get this passing, we need to make the test suite ignore the techpreview warning from Kube API Server, I've got a PR for that here https://github.com/openshift/origin/pull/26349

**Comment by JoelSpeed** [comment]
Latest results show that the storage issues are gone now the rebase has landed. But does now highlight that there are two tests that fail because of alerts fired because we are running as TechPreview. I'll raise PRs to follow up on getting those alerts ignored when in tech preview mode

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml
Comment (EmilienM) [inline ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:267]: I thought we had a job already for that:
https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/e2e/openstack/ccm/openshift-e2e-openstack-ccm-workflow.yaml

@mdbooth @mandre

Comment (JoelSpeed) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9.yaml:267]: That one is specific for CCM, this PR is adding generic `TechPreviewNoUpgrade` tests to prove out TP features in general and in the future, CCM just happens to be a subset of that for now

**Comment by JoelSpeed** [comment]
Test failures now seem to be unrelated to the TechPreview testing and may actually be genuine problems/flakes. I think this is ready to go in now.

The Azure and OpenStack serials have passed already which is a good sign :) 

One more test for luck
/retest

File: ci-operator/step-registry/ipi/conf/azure/techpreview/ipi-conf-azure-techpreview-chain.metadata.json
Comment (wking) [inline ci-operator/step-registry/ipi/conf/azure/techpreview/ipi-conf-azure-techpreview-chain.metadata.json:5]: I guess inherited from `master`, but ideally each directory will have more than one approver.  Can we find anyone else who can sign up to help maintain this chain going forward?

File: ci-operator/step-registry/ipi/conf/openstack/techpreview/ipi-conf-openstack-techpreview-chain.yaml
Comment (wking) [inline ci-operator/step-registry/ipi/conf/openstack/techpreview/ipi-conf-openstack-techpreview-chain.yaml:12]: I feel like there should be a chain gathering up some of this, to reduce the chance that someone adds a new OpenStack knob similar to the existing FIPS, but forgets to update this tech-preview chain to respect that knob here.

**Comment by stbenjam** [comment]
Looks like you'll need someone like @jupierce or @bradmwilliams for final approval

File: ci-operator/step-registry/ipi/conf/openstack/techpreview/ipi-conf-openstack-techpreview-chain.yaml
Comment (JoelSpeed) [reply ci-operator/step-registry/ipi/conf/openstack/techpreview/ipi-conf-openstack-techpreview-chain.yaml:12]: There doesn't appear to be anything today that I can find that resembles a shared chain for this stuff, have poked the ShiftStack team regarding the issue to see if this is something on their radar

Comment (JoelSpeed) [reply ci-operator/step-registry/ipi/conf/openstack/techpreview/ipi-conf-openstack-techpreview-chain.yaml:12]: @pierreprinetti suggested that we can actually avoid this duplcation by not using the ipi-conf like this and just directly adding the techpreview step into the workflow. Since the AWS version of this is already merged, https://github.com/openshift/release/pull/21227 is a PR which cleans it up to show what Pierre meant, do you think this is a better approach that I should adopt for all of these?

**Comment by JoelSpeed** [comment]
Just rebased to fix conflicts, someone updated the master periodics to move them from build01 to build02

**Review by bradmwilliams — APPROVED** [review]

**Pull request merged**
